### DISCUSSION TITLE
Getting value with Generic events

### DISCUSSION THREAD

**Discussion by DMazykin** [zauberzeug/nicegui#1875] [Q&A]
### Description

I can't manage to get a value of field with "blur" generic event.
Here is my code:

```py
from nicegui import ui


def notify(event):
    ui.notify(event)


ui.number(label="Number with blur", value=3.1415927, format="%.2f").on("blur", notify)


ui.number(label="Number with change", value=3.1415927, format="%.2f").on(
    "change", notify
)

ui.number(
    label="Number with lambda",
    value=3.1415927,
    on_change=lambda e: notify(e),
    format="%.2f",
)
result = ui.label()

ui.run()
```

Running this code I getting the following output:

**"Number with blur":**
`
GenericEventArguments(sender=<nicegui.elements.number.Number object at 0x7f02c883f4d0>, client=<nicegui.client.Client object at 0x7f02c8d2f2d0>, args={'isTrusted': True, '_vts': 1698185494648, 'detail': 0, 'sourceCapabilities': {}, 'which': 0, 'type': 'focusout', 'currentTarget': None, 'eventPhase': 0, 'bubbles': True, 'cancelable': False, 'defaultPrevented': False, 'composed': True, 'timeStamp': 40745, 'returnValue': True, 'cancelBubble': False, 'NONE': 0, 'CAPTURING_PHASE': 1, 'AT_TARGET': 2, 'BUBBLING_PHASE': 3})
`

**"Number with change"**
`
GenericEventArguments(sender=<nicegui.elements.number.Number object at 0x7f02ec572150>, client=<nicegui.client.Client object at 0x7f02c8d2f2d0>, args='3.14')
`

**"Number with lambda"**
`
ValueChangeEventArguments(sender=<nicegui.elements.number.Number object at 0x7f02ec576350>, client=<nicegui.client.Client object at 0x7f02c8d2f2d0>, value=3.14)
`

**Comment by falkoschindler** [comment]
You can always access the `sender` of the event, which is the input with a `value`:
```py
ui.number("Number with blur").on("blur", lambda e: ui.notify(e.sender.value))
```
